base/uk.ac.stfc.isis.ibex.ui/src/uk/ac/stfc/isis/ibex/ui/tables/DataboundTable.java

```diff
@@ -235,6 +246,24 @@ public void removeMouseListener(MouseListener listener) {
 		table.addMouseListener(listener);
 	}
 	
+	/**
+	 * Sets the a listener which will be run whenever the items of the table 
+	 * are sorted.
+	 * @param runnable the new sort listener of the table.
+	 */
+	public void setSortListener(Runnable runnable) {
+	    sortListener = runnable;
+	}
+	
+	/**
+     * Sets the a listener which will be run whenever items are added to or
+     * removed from the table.
+     * @param runnable the new change listener of the table.
+     */
+	public void setChangeListener(Runnable runnable) {
+	    changeListener = runnable;
+	}
+	
```

Comment:
I don't really like these listeners as:
* They're not like normal databinding listeners as they're just plain runnables
* We're having to handle how they're called ourselves which means in the case of the sort we have a listener calling a listener which feels odd.

I think the change one can be removed with the above code and I think the sort one would be better if it was just an additional listener added to the column.

Reply:
It would have been achievable, but would have needed a new subclass of DataboundTable that the tables changed in this PR would inherit from, or make a new getSortListener method in DataboundTable that had a sensible default implementation. It would have taken a while and I decided it is not worth it.

I have made the sort Listener an optional that wraps a runnable.